src/datastation/common/utils.py

```diff
@@ -19,6 +20,12 @@ def add_batch_processor_args(parser, report: bool = True):
                             dest='report_file')
 
 
+def raise_for_status(r):
```

Comment:
I may need another name to avoid confusion. We may need to log at level info or warning for (some) status codes < 400.